jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1736,12 +1737,17 @@ ecma_builtin_typedarray_prototype_index_helper (ecma_value_t this_arg, /**< this
       return ECMA_VALUE_ERROR;
     }
 
-    from_index = ecma_builtin_helper_array_index_normalize (num_var, length, is_last_index_of);
-  }
-
-  if (!ecma_is_value_number (args[0]))
-  {
-    return ecma_make_integer_value (-1);
+    if (!ecma_number_is_nan (num_var)
+        && is_last_index_of
+        && num_var < 0
+        && (uint32_t) abs((int) num_var) > length)
```

Comment:
Not really, because if the second argument is negative, than the normalized index is calculated this way: norm_index = length - abs (from_index). And since we return with uint32_t in the array_index_normalize method, if this negative argument's absolute value is greater than the length, the norm_index itself becomes negative, and this will result an underflow. So if we know, that the second argument is negative, and its absolute value is greater then the length, we should return -1, but if this absolute value is less then the length, we could calculate with it.